apps/condo/domains/marketplace/schema/MarketCategory.js

```diff
@@ -0,0 +1,124 @@
+const { GQLListSchema, getById } = require('@open-condo/keystone/schema')
+
+const { getGQLErrorValidator } = require('@condo/domains/common/schema/json.utils')
+const FileAdapter = require('@condo/domains/common/utils/fileAdapter')
+const { getFileMetaAfterChange } = require('@condo/domains/common/utils/fileAdapter')
+const access = require('@condo/domains/marketplace/access/MarketCategory')
+const { ERROR_INVALID_MOBILE_SETTINGS } = require('@condo/domains/marketplace/constants')
+
+const ERRORS = {
+    MAXIMUM_DEPTH_REACHED: {
+        code: BAD_USER_INPUT,
+        type: 'MAXIMUM_DEPTH_REACHED',
+        message: 'The depth of category inheritance should be no more than 2',
+        messageForUser: 'api.marketplace.MarketCategory.MAXIMUM_DEPTH_REACHED',
+    },
+    CANNOT_CONNECT_TO_ITSELF: {
+        code: BAD_USER_INPUT,
+        type: 'CANNOT_CONNECT_TO_ITSELF',
+        message: 'Cannot connect to itself',
+        messageForUser: 'api.marketplace.MarketCategory.CANNOT_CONNECT_TO_ITSELF',
+    },
+}
+
+const MARKET_CATEGORY_FILE_ADAPTER = new FileAdapter('market_category')
+const imageMetaAfterChange = getFileMetaAfterChange(MARKET_CATEGORY_FILE_ADAPTER, 'image')
+
+const ajv = new Ajv()
+addFormats(ajv)
+
+const settingsFieldSchema = {
+    type: 'object',
+    additionalProperties: false,
+    required: ['bgColor', 'titleColor'],
+    properties: {
+        bgColor: {
+            type: 'string',
+            pattern: '^#(?:[0-9a-fA-F]{3}){1,2}$',
+        },
+        titleColor: {
+            type: 'string',
+            pattern: '^#(?:[0-9a-fA-F]{3}){1,2}$',
+        },
+    },
+}
+const validateSettingsField = getGQLErrorValidator(ajv.compile(settingsFieldSchema), ERROR_INVALID_MOBILE_SETTINGS)
+
+const MarketCategory = new GQLListSchema('MarketCategory', {
+    schemaDoc: 'An entity that allows you to define a category tree in the marketplace',
+    adminDoc: 'mobileSettings = {"bgColor":"#rrggbb", "titleColor":"#rrggbb"}',
+    fields: {
+
+        name: {
+            schemaDoc: 'Key for i18n',
+            type: LocalizedText,
+            isRequired: true,
+            template: 'marketplace.marketCategory.*.name',
+        },
+
+        image: {
+            schemaDoc: 'Preview image',
+            type: File,
+            isRequired: true,
+            adapter: MARKET_CATEGORY_FILE_ADAPTER,
+        },
+
+        mobileSettings: {
+            schemaDoc: 'Settings for mobile ui',
+            type: Json,
+            isRequired: true,
+            hooks: {
+                validateInput: validateSettingsField,
+            },
+        },
+
+        parentCategory: {
+            schemaDoc: 'Which category does this subcategory belong to',
+            type: Relationship,
+            ref: 'MarketCategory',
+            kmigratorOptions: { null: true, on_delete: 'models.CASCADE' },
+            hooks: {
+                afterChange: async (args) => {
+                    await imageMetaAfterChange(args)
```

Comment:
Why is meta update not called in schema hook?

Reply:
misclicked) corrected)